src/main/java/org/spongepowered/server/launch/plugin/PluginCandidate.java

```diff
@@ -264,13 +273,35 @@ private boolean verifyVersionRange(String id, @Nullable String expectedRange, @N
 
                     this.versions.put(id, expectedRange);
 
+                    if (range.getRecommendedVersion() instanceof DefaultArtifactVersion) {
+                        BigInteger majorExpected = ((DefaultArtifactVersion) range.getRecommendedVersion()).getVersion().getFirstInteger();
+                        if (majorExpected != null) {
+                            BigInteger majorInstalled = installedVersion.getVersion().getFirstInteger();
+
+                            // Show a warning if the major version does not match,
+                            // or if the installed version is lower than the recommended version
+                            if (majorInstalled != null
+                                    && (!majorExpected.equals(majorInstalled) || installedVersion.compareTo(range.getRecommendedVersion()) < 0)) {
+                                VanillaLaunch.getLogger().warn("Plugin {} from {} was designed for {} {}. It may not work properly.",
```

Comment:
A lot of the warnings/errors in SpongeVanilla could use some improvements (some could use additional information, others should be likely errors instead of warnings). We can improve all these messages together at a later point - for now this should do the job.